yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -727,6 +712,44 @@ export class ProvingOrchestrator implements BlockProver {
     );
   }
 
+  // Enqueues the tub circuit for a given transaction index
+  // Once completed, will enqueue the next circuit, either a public kernel or the base rollup
+  private enqueueTube(provingState: ProvingState, txIndex: number) {
```

Comment:
I'm confused here: doesn't `enqueueFirstProofs` handle enqueuing the public kernels at the end, after calling this `enqueueTube`?

```
    // Enqueue all of the VM/kernel proving requests
    // Rather than handle the Kernel Tail as a special case here, we will just handle it inside enqueueVM
    for (let i = 0; i < numPublicKernels; i++) {
      logger.debug(`Enqueueing public VM ${i} for tx ${txIndex}`);
      this.enqueueVM(provingState, txIndex, i);
    }
```

Reply:
No, that enqueues all of the VM proofs. 

The tube and all VM proofs can be proven concurrently. Then, the kernels can be proven as and when their individual dependencies have been completed.

Reply:
Got it, thanks!